handler/interceptors.go

```diff
@@ -103,13 +103,28 @@ func (interceptor *paymentValidationInterceptor) intercept(srv interface{}, ss g
 	}
 	log.Debug("Payment validated")
 
-	e := handler(srv, ss)
+	handlerSucceed := false
+
+	defer func() {
+		if !handlerSucceed {
+			if r := recover(); r != nil {
+				e = r.(error)
+				paymentHandler.CompleteAfterError(payment, e)
+				panic("re-panic after payment handler error handling")
+			} else if e != nil {
+				err = paymentHandler.CompleteAfterError(payment, e)
+				if err != nil {
+					e = err.Err()
+				}
+			}
+		}
+	}()
+
+	e = handler(srv, ss)
+	handlerSucceed = err != nil
```

Comment:
Should it be?
```suggestion
	handlerSucceed = e != nil
```
Or we could move ```handlerSucceed = true``` to the line 130

Reply:
Yes, it should check e. I have updated this according to the suggestion.